packages/js-lib/src/validation/validation.ts

```diff
@@ -15,12 +15,12 @@ import type { AppError } from '../error/error.util.js'
  *
  * @experimental
  */
-export type ValidationFunction<T, ERR extends AppError> = (
-  input: T,
+export type ValidationFunction<IN, OUT, ERR extends AppError> = (
```

Comment:
I will propose to try and make OUT = IN by default, which would need to use provide a default for ERR too, which can be `AppError`